### PR TITLE
Set up Jaeger proto

### PR DESCRIPTION
**Pull request by guilload**
### Description
Commit Jaeger proto and generate stubs

### How was this PR tested?
```
cargo build --manifest-path quickwit/Cargo.toml -p quickwit-proto
```

### CODE REVIEW

File: quickwit/quickwit-proto/.gitignore [hunk 1]
```diff
@@ -0,0 +1,2 @@
+gogoproto.rs
```
Comment (guilload) [inline quickwit/quickwit-proto/.gitignore:1]: Those are empty files generated by tonic.

File: quickwit/quickwit-proto/Cargo.toml [hunk 2]
```diff
@@ -11,9 +11,11 @@ documentation = "https://quickwit.io/docs/"
 
 [dependencies]
 prost = { workspace = true }
+prost-types = { workspace = true }
```
Comment (guilload) [inline quickwit/quickwit-proto/Cargo.toml:14]: Protocol Buffers well known types.

File: quickwit/quickwit-proto/build.rs [hunk 3]
```diff
@@ -17,25 +17,16 @@
 // You should have received a copy of the GNU Affero General Public License
 // along with this program. If not, see <http://www.gnu.org/licenses/>.
 
-fn main() -> Result<(), Box<dyn std::error::Error>> {
-    println!("cargo:rerun-if-changed=proto/ingest_api.proto");
```
Comment (guilload) [inline quickwit/quickwit-proto/build.rs:21]: Those are actually already generated by `tonic::compile`.  You can see them when running `cargo` in verbose mode `-vv`.

File: quickwit/quickwit-proto/build.rs [hunk 4]
```diff
@@ -58,16 +49,63 @@ fn main() -> Result<(), Box<dyn std::error::Error>> {
...
+        .ok()?
+        .flatten()
+        .map(|dir_entry| dir_entry.file_name())
+        .sorted()
+        .last()
+    {
+        let protoc_include_dir = Path::new("/opt/homebrew/Cellar/protobuf")
+            .join(protoc_version)
+            .join("include");
+        println!(
```
Comment (guilload) [inline quickwit/quickwit-proto/build.rs:103]: `cargo` swallows the print, unless `-vv`.

File: quickwit/quickwit-proto/build.rs [hunk 5]
```diff
@@ -49,25 +45,71 @@ fn main() -> Result<(), Box<dyn std::error::Error>> {
...
+    } else {
+        None
+    }
+}
+
+fn find_jaeger_includes() -> Vec<PathBuf> {
+    let mut includes = vec![PathBuf::from("./jaeger")];
+
+    if let Ok(protoc_include_dir) = env::var("QW_PROTOC_INCLUDE_DIR") {
+        includes.push(PathBuf::from(protoc_include_dir));
```
Comment (fulmicoton) [inline quickwit/quickwit-proto/build.rs:102]: That little dance is for google well-known types right?
I suggest we remove all this logic and just copy the google protos in the repo. 
They are not more sacred than the jaeger protos imho, and our build process will be less brittle
and less environment dependent.

**Review by fulmicoton — APPROVED** [review]
Approved but I think we'd better just include google protos.

**Pull request merged**